search/includes/classes/commands/class-corecommand.php

```diff
@@ -68,9 +68,33 @@ public function index( $args, $assoc_args ) {
 
 		$this->_maybe_setup_index_version( $assoc_args );
 
-		array_unshift( $args, 'elasticpress', 'index' );
+		/**
+		 * EP's `--network-wide` mode uses switch_to_blog to index the content,
+		 * that may not be reliable if the codebase differs between subsites.
+		 *
+		 * Side-step the issue by spawning child proccesses for each subsite.
+		 */
+		if ( isset( $assoc_args['network-wide'] ) && is_multisite() ) {
+			$start = microtime( true );
+			WP_CLI::line( 'Operating in network mode!' );
+
+			unset( $assoc_args['network-wide'] );
+
+			foreach ( get_sites() as $site ) {
+				switch_to_blog( $site->blog_id );
+				$assoc_args['url'] = home_url();
+
+				WP_CLI::line( 'Indexing ' . $assoc_args['url'] );
+				WP_CLI::runcommand( 'vip-search index ' . Utils\assoc_args_to_str( $assoc_args ) );
+				Utils\wp_clear_object_cache();
+			}
 
-		WP_CLI::run_command( $args, $assoc_args );
+			WP_CLI::line( WP_CLI::colorize( '%CNetwork-wide run took: ' . ( round( microtime( true ) - $start, 3 ) ) . '%n' ) );
+			restore_current_blog();
```

Comment:
This `restore_current_blog()` should be matched to the `switch_to_blog()` (i.e. inside the `foreach`) so that the global array that tracks the stack of switched sites, and the global `switched` is correct.

Reply:
Nice catch, thanks!